### PR TITLE
docs: first brush over the geometry pages

### PR DESCRIPTION
**Pull request by asalzburger**
This PR combines the two geometry pages into one and adds some additional info and highlights.

### CODE REVIEW

File: docs/core/geometry.md [hunk 1]
```diff
@@ -209,41 +223,57 @@ Possible extensions are:
...
 geometry,
 
-## Attaching a 3D detector geometry
+### Attaching a 3D detector geometry
 
 Usually, a 3D detector model geometry exists, which is either native to the full
 detector simulation (Geant4) or is translated into it. This model, however, is
 in general too detailed for track reconstruction: navigating through the
-detailed detector geometry
+detailed detector geometry is generally costy and one can profit greatly from a simplication mechanism.
```
Comment (paulgessinger) [inline docs/core/geometry.md:244]:
```suggestion
detailed detector geometry is generally costly and one can profit greatly from a simplication mechanism.
```

File: docs/core/geometry.md [hunk 2]
```diff
@@ -126,6 +131,10 @@ surfaces, a special `InfiniteBounds` class is available.
 ![DiscBounds](/figures/geometry/DiscBounds.png)
 ![PlaneBounds](/figures/geometry/PlaneBounds.png)
 
+```{tip}
+The coordinate systems define - in an ideal setup - also the readout measurement directions. In such a case, a track prediction from the propgation, e.g. will already be in the correct frame of the measurement and residual or compatibility checks will not need additional coordinate transformations.
```
Comment (AJPfleger) [inline docs/core/geometry.md:135]:
```suggestion
The coordinate systems define - in an ideal setup - also the readout measurement directions. In such a case, a track prediction from the propagation, e.g. will already be in the correct frame of the measurement and residual or compatibility checks will not need additional coordinate transformations.
```

File: docs/core/geometry.md [hunk 3]
```diff
@@ -184,7 +193,7 @@ at creation.
 ![VolumeBounds](/figures/geometry/VolumeBounds.png)
 ![CylinderVolumeBounds](/figures/geometry/CylinderVolumeBounds.png)
 
-## Material description
+## Detector Material description
```
Comment (AJPfleger) [inline docs/core/geometry.md:196]:
```suggestion
## Detector material description
```

File: docs/core/geometry.md [hunk 4]
```diff
@@ -197,6 +206,11 @@ The basic information for any material is:
 
 This information is confined together in the `Material` class.
 
+```{note}
+In track reconstruction, only an effective material description is needed, i.e. non-physical values in regards of atomic number, elementary charge or even density are allowed, as long as the effective relative radiation length and `A/Z x rho` ratio can be retrieved.
```
Comment (AJPfleger) [inline docs/core/geometry.md:210]:
```suggestion
In track reconstruction, only an effective material description is needed, i.e. non-physical values in regards of the atomic number, the elementary charge or even the density are allowed, as long as the effective relative radiation length and `A/Z x rho` ratio can be retrieved.
```

File: docs/core/geometry.md [hunk 5]
```diff
@@ -209,41 +223,57 @@ Possible extensions are:
  * `HomogeneousSurfaceMaterial`, homogeneous material description on a surface
  * `BinnedSurfaceMaterial`, an arbitrarily binned material description with a
     corresponding `BinUtility` object
+
+In addition, a dedicated extension exists to allow configuration of the material
+mapping process, that is in further descibed below.
```
Comment (AJPfleger) [inline docs/core/geometry.md:228]:
```suggestion
mapping process, that is in further described below.
```

File: docs/core/geometry.md (hunk 1 above)
Comment (AJPfleger) [reply docs/core/geometry.md:244]: also "simplification" instead "simplication"?

**Review by AJPfleger — CHANGES_REQUESTED** [review]

File: docs/core/geometry.md [hunk 6]
```diff
@@ -209,41 +223,57 @@ Possible extensions are:
...
-### Existing plugins for 3D geometry libraries
+#### Existing plugins for 3D geometry libraries
 
 Very simple helper methods for 3D libraries exist, they are certainly not
 optimised, but used for templating:
 
-* `TGeoDetectorElement` connects a TGeo volume to a `Surface` (see [](TGeo plugin))
-* `DD4HepDetectorElement` connects a DD4hep volume (based on TGeo) to a `Surface` (see [](DD4hep plugin))
+* `TGeoDetectorElement` connects a TGeo volume to a `Surface`
+* `DD4HepDetectorElement` connects a DD4hep volume (based on TGeo) to a `Surface`
```
Comment (benjaminhuth) [inline docs/core/geometry.md:262]: We could add the `Geant4DetectorElement` here as well, maybe also with a link to the documentation (via the `{class}` prefix and namespace)

**Review by benjaminhuth — APPROVED** [review]
I think we can also merge this in in the current state, right?

File: docs/core/geometry.md (hunk 6 above)
Comment (asalzburger) [reply docs/core/geometry.md:262]: This will have to be updated anyway for the `Detector` infrastructure, I will make a new PR once this is done.

**Comment by asalzburger** [comment]
> I think we can also merge this in in the current state, right?

Yes, I did a little cleanup & added the `Geant4DetectorElement` as suggested.

**Review by benjaminhuth — APPROVED** [review]

**Pull request merged**